src/parser.rs

```diff
@@ -444,28 +444,24 @@ impl Parser {
         };
         let on_overflow = if self.parse_keywords(vec!["ON", "OVERFLOW"]) {
             let error = self.parse_keyword("ERROR");
-            let filler = if !error {
-                self.expect_keyword("TRUNCATE")?;
-                Some(Box::new(self.parse_expr()?))
+            Some(if error {
```

Comment:
We avoid wrapping long expressions in Some. Please use `Some(ListAggOnOverflow::Error)` instead.